### PR TITLE
crypto components

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- CryptoComponents 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/node/config/testKeys/validatorKey.pem [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-----BEGIN PRIVATE KEY for 75907bf66c64949f8d1dd003ed8fd815e8dec189a9e50e4bfd7e2592cfdc92641baf500e3533820e0f58c4ed3f39c01750f0017726731fe5bc66a0ab6fc9a7b5661d163e72f099fe29df16fd59ded198dc1423575be1b6e2160e0d84a5ece08a-----
```
Comment (iulianpascalau) [inline cmd/node/config/testKeys/validatorKey.pem:1]: I think we need another approach here.
We can leave it as it is and I will fix this in an upcoming PR

File: node/processingOnlyNode/cryptoComponents.go [hunk 2]
```diff
@@ -1,3 +1,238 @@
...
+	"github.com/multiversx/mx-chain-go/common"
+	cryptoCommon "github.com/multiversx/mx-chain-go/common/crypto"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/factory"
+	cryptoComp "github.com/multiversx/mx-chain-go/factory/crypto"
+	"github.com/multiversx/mx-chain-go/vm"
+)
+
+type ArgsCryptoComponentsHolder struct {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply node/processingOnlyNode/cryptoComponents.go:18]: done

File: node/processingOnlyNode/cryptoComponents.go [hunk 3]
```diff
@@ -1,3 +1,240 @@
...
+	return c.managedPeersHolder
+}
+
+// KeysHandler will return the keys handler
+func (c *cryptoComponentsHolder) KeysHandler() consensus.KeysHandler {
+	return c.keysHandler
+}
+
+// Clone will clone the cryptoComponentsHolder
+func (c *cryptoComponentsHolder) Clone() interface{} {
```
Comment (bogdan-rosianu) [inline node/processingOnlyNode/cryptoComponents.go:215]: why not return a `*cryptoComponentsHolder` ?

Comment (miiu96) [reply node/processingOnlyNode/cryptoComponents.go:215]: I had to implement the interface

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**